firestore/client_test.go

```diff
@@ -32,6 +35,61 @@ var testClient = &Client{
 	readSettings: &readSettings{},
 }
 
+func TestNewClientWithDatabase(t *testing.T) {
+	for _, tc := range []struct {
+		desc       string
+		databaseID string
+		projectID  string
+		wantErr    bool
+	}{
+		{
+			desc:       "Empty databaseID",
+			databaseID: "",
+			projectID:  "p1",
+			wantErr:    true,
+		},
+		{
+			desc:       "Error from NewClient bubbled to NewClientWithDatabase",
+			databaseID: "db1",
+			projectID:  "",
+			wantErr:    true,
+		},
+		{
+			desc:       "Valid databaseID",
+			databaseID: "db1",
+			projectID:  "p1",
+			wantErr:    false,
+		},
+	} {
+
+		srv, cleanup, err := newMockServer()
+		if err != nil {
+			t.Fatalf("NewClientWithDatabase: Failed to create mock server %v", err)
+		}
+
+		conn, err := grpc.Dial(srv.Addr, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+		if err != nil {
+			t.Fatalf("NewClientWithDatabase: Failed to create connection %v", err)
+		}
+
+		client, err := NewClientWithDatabase(context.Background(), tc.projectID, tc.databaseID, option.WithGRPCConn(conn))
```

Comment:
Shall we just assert against `client.databaseID` here and avoid all the gRPC setup?

Reply:
Done